entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy

```diff
@@ -36,7 +36,7 @@ declare private function es-codegen:casting-function-name(
 {
     if ($datatype eq "iri")
     then "sem:iri"
-    else "xs:" || $datatype
+    else "xs:" || $datatype || "#1"
```

Comment:
In XQuery, one refers to a function literal with the name of the function, and the number of arguments appended.

So map:map#3  is one function.  A casting function has one arg, hence #1